src/cli/embedded_help.rb

```diff
@@ -1,3 +1,11 @@
+module EmbeddedScripting
+  @@fileNames = EmbeddedScripting.allFileNamesAsString.split(';')
+
+  def self.fileNames
+    @@fileNames
+  end
+end
+
```

Comment:
No problem here. It originally pained me a little that we would have to go through the process of joining strings in C++ with ";" then split in ruby, instead of just properly wrapping `std::vector<std::string>` to use `fileNames()` directly. We do that for eg https://github.com/NREL/OpenStudio/blob/70bfd7a226f0da606ced48db8f74107cf22f0af5/src/utilities/core/CommonImport.i#L24

So I tried to do that... But as soon at you import `<std_vector.i>` suddenly you catch `multiple definition of multiple definition of swig::GC_VALUE::hash_id` etc.
That's because swig internally does define it (in ruby.swg) and that's used when using the STL. But we do shenanigans in main.cpp (and GC_Value.hpp) and duplicate that stuff. Since I couldn't see an easy way around that, I'm totally fine with your solution

Reply:
There seems to be an interesting story here. 

Dan made this comment 5 years ago: https://github.com/NREL/OpenStudio/blame/70bfd7a226f0da606ced48db8f74107cf22f0af5/src/cli/EmbeddedScripting.i#L7

And he's right, if you include the vector support and remove all of our own definitions for the `GC_VALUE` code, it crashes, and I have no idea why.

Then regarding this very poorly performing `;` concatenation followed by `;` parsing we were doing (1600+ times while loading we did both things!)

Two years later Dan then commented that he didn't know why `fileNames()` wasn't swig'd correctly

https://github.com/NREL/OpenStudio/blame/70bfd7a226f0da606ced48db8f74107cf22f0af5/src/cli/embedded_help.rb#L577-L579

I decided it was much more expedient to just do the work in Ruby then try to understand something that probably comes down to a linking issue regarding the crash mentioned in the first commit log.